lib/api/upload.js

```diff
@@ -0,0 +1,114 @@
+/**
+  * @module storj/upload
+  * @license LGPL-3.0
+  */
+
+'use strict';
+
+var EncryptStream = require('storj-lib/lib/crypto-tools/encrypt-stream.js');
+var crypto = require('crypto');
+var storjUtils = require('storj-lib/lib/utils');
+var through = require('through');
+
+var DeterministicKeyIv =
+  require('storj-lib/lib/crypto-tools/deterministic-key-iv');
+
+var uploadMeta = {};
+
+module.exports = function(bucketId, fileName, opts, cb) {
```

Comment:
Lets attach a name to this function for meaningful stack traces.

I need to add a linting rule for this because I'm sure there are other places in the code where I'm not doing this.